src/AdoNet/Shared/Storage/OrleansRelationalDownloadStream.cs

```diff
@@ -203,7 +203,7 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             catch(Exception e)
             {
                 //Due to call to Read, this is for sure a IOException and can be thrown out.
-                var tcs = new TaskCompletionSource<int>();
+                var tcs = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
                 tcs.SetException(e);
```

Comment:
This actually makes no sense in this context. You're not in an async context, and the task you are returning is already completed. You should opt for using `Task.FromException()` and `Task. FromCanceled`

Reply:
Well spotted. The PR was mostly find and replace. When these methods were originally written, FromException and FromCanceled didn't exist (neither did Task.CompletedTask), hence the TCS.

Would you be willing to PR your suggestion?

Reply:
Yeah, I figured as much :-)
I can do

Reply:
Thank you, @icanhasjonas, I appreciate the effort